common/djangoapps/course_modes/tests/test_views.py

```diff
@@ -511,64 +516,118 @@ def test_course_closed(self):
             redirect_url = reverse('dashboard') + '?course_closed=1%2F1%2F15%2C+12%3A00+AM'
             self.assertRedirects(response, redirect_url)
 
-    # Value Prop TODO (REV-2378): remove waffle flag from tests once the new Track Selection template is rolled out.
-    # Other tests may need to be updated/removed to reflect the new page.
-    # The below test can be separated into multiple tests once un-happy path is implemented.
-    def test_new_track_selection(self):
-        # For the new track selection template to render, FBE must be fully on (gated_content and audit_access_deadline)
-        # and happy path conditions must be met:
-        # User can upgrade, FBE is fully on, and user is not an enterprise user.
+    def _assert_fbe_page(self, response, min_price=None, **_):
+        """
+        Assert fbe.html was transcluded.
+        """
+        self.assertContains(response, "Choose a path for your course in")
+
+        # Check if it displays the upgrade price for verified track and "Free" for audit track
+        self.assertContains(response, min_price)
+        self.assertContains(response, "Free")
+
+        # Check for specific HTML elements
+        self.assertContains(response, '<span class="award-icon">')
+        self.assertContains(response, '<span class="popover-icon">')
+        self.assertContains(response, '<span class="note-icon">')
+
+        # Check for upgrade button ID
+        self.assertContains(response, 'track_selection_upgrade')
+
+        # Check for audit button ID
+        self.assertContains(response, 'track_selection_audit')
+
+        # Check for happy path messaging - verified
+        self.assertContains(response, '<li class="collapsible-item">')
+        self.assertContains(response, 'access to all course activities')
+        self.assertContains(response, 'Full access')
+
+        # Check for happy path messaging - audit
+        self.assertContains(response, "discussion forums and non-graded assignments")
+        self.assertContains(response, "Get temporary access")
+        self.assertContains(response, "Access expires and all progress will be lost")
+
+    def _assert_unfbe_page(self, response, min_price=None, **_):
+        """
+        Assert track_selection.html and unfbe.html were transcluded.
+        """
+        # Check for string unique to track_selection.html.
+        self.assertContains(response, "| Upgrade Now")
+        # This string only occurs in lms/templates/course_modes/track_selection.html
+        # and related theme and translation files.
+
+        # Check for string unique to unfbe.html.
+        self.assertContains(response, "Some graded content may be lost")
+        # This string only occurs in lms/templates/course_modes/unfbe.html
+        # and related theme and translation files.
+
+        # Check min_price was correctly passed in.
+        self.assertContains(response, min_price)
```

Comment:
Is there a reason the `min_price` is checked on all 3 places (main test, and the fbe/unfbe)?

Reply:
I think there are only 2 references to `min_price` in this PR. What happened was this PR removed `test_new_track_selection()` and moved its asserts to `_assert_fbe_page()`.

Since `test_new_track_selection()` previously performed a check on `min_price`, `_assert_fbe_page()` contains a check on `min_price`.

However, this does not test the unhappy path pages for `min_price`. Thus, `_assert_unfbe_page()` has an assert of `min_price()` as well.